src/npc_attack.cpp

```diff
@@ -384,30 +470,30 @@ bool npc_attack_throw::can_use( const npc &source ) const
     if( single_item.count_by_charges() ) {
         single_item.charges = 1;
     }
-    // please don't throw your pants...
-    return !source.is_worn( thrown_item ) &&
-           // we would rather throw than activate this. unless we need to throw it now.
-           !( thrown_item.has_flag( flag_NPC_THROW_NOW ) || thrown_item.has_flag( flag_NPC_ACTIVATE ) ) &&
-           // i don't trust you as far as i can throw you.
-           source.throw_range( single_item ) != 0;
+    if( thrown_item.has_flag( flag_NPC_THROW_NOW ) || thrown_item.has_flag( flag_NPC_THROWN ) ) {
+        //Always allow throwing items that are flagged as throw now or npc_thrown
+        return true;
+    }
+
+    bool throwable = source.throw_range( single_item ) > 0 && !source.is_worn( thrown_item ) &&
+                     !thrown_item.has_flag( flag_NPC_ACTIVATE );
+    throwable = throwable && !thrown_item.is_gun() && !thrown_item.is_armor() &&
+                !thrown_item.is_comestible() && !thrown_item.is_magazine() && !thrown_item.is_tool() &&
+                !thrown_item.is_unarmed_weapon();
```

Comment:
seems really wonky tbh. we really should have some better logic here that doesn't require you to check the item's type in code like this.

Reply:
Yup. Left a comment below so it can be addressed eventually.